app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -170,9 +177,22 @@ def process_collection(collection, key)
         end
       end
 
-      def gather_data_for_this_region(arm_service, method = "list")
-        @data[:resource_groups].collect do |resource_group|
-          arm_service.send(method, resource_group[:name])
+      # Gather all data for all resource groups. If a subscription_id is provided,
+      # then only collect data for that subscriptions. Otherwise, collect alll data
+      # for all subscriptions.
+      #
+      def gather_data_for_this_region(arm_service, method = 'list_all', subscription_id = nil)
+        subscriptions = subscription_id ? [subscription_id] : @subscriptions
+
+        subscriptions.collect do |sub_id|
+          arm_service.configuration.subscription_id = sub_id
+          if method.to_s == 'list_all'
+            arm_service.send(method)
+          else
+            @data[:resource_groups].collect do |resource_group|
+              arm_service.send(method, resource_group[:name])
+            end
+          end
         end.flatten
       end
```

Comment:
@djberg96 - Azure will not give us any inventory if the subscription and tenant IDs do not match so I suggest we skip over them and log the exception message which happens to be quite helpful